File: src/main/java/br/com/kronos/backend/aplicacao/basecurricular/Objetivo.java
package br.com.kronos.backend.aplicacao.basecurricular;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Objetivo {
    
    private long id;
    private Long idCampoExperiencia;
    private String descricao;
    private String codigo;
    private boolean ativo;
    private Boolean bncc;
    private Long idInstituicao;
    private Long idFaixaAno;

	@Builder
    public Objetivo(long id, Long idCampoExperiencia, String descricao, String codigo, boolean ativo, Boolean bncc, 
                    Long idInstituicao, Long idFaixaAno) {

		super();
        this.id = id;
        this.idCampoExperiencia = exigirNaoNulo(idCampoExperiencia, "Campo de experiência"); 
        this.descricao = exigirTamanho(descricao, "Descrição", 1, 400);   
        this.codigo = exigirTamanho(codigo, "Código", 1, 10); 
        this.ativo = exigirNaoNulo(ativo, "Ativo");
        if (bncc != null && bncc) {
        	this.bncc = bncc;
        	this.idInstituicao = null; 
        } else {
        	this.bncc = false;
        	this.idInstituicao = idInstituicao;     
        }
        this.idFaixaAno = exigirNaoNulo(idFaixaAno, "Faixa / Ano");    
	}
}

File: poc-kronos-frontend/src/model/basecurricular/Objetivo.js
class Objetivo {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.idCampoExperiencia = dados ? dados.idCampoExperiencia : null;
    this.camposExperiencia = dados ? dados.camposExperiencia : null;
    this.descricao = dados ? dados.descricao : null;
    this.codigo = dados ? dados.codigo : null;
    this.ativo = dados ? dados.ativo : null;
    this.bncc = dados ? dados.bncc : null;
    this.idFaixaAno = dados ? dados.idFaixaAno : null;
    this.faixaAno = dados ? dados.faixaAno : null;
  }

  static validations(required, maxLength) {
    return {
      descricao: { required, maxLength: maxLength(400) },
      codigo: { required, maxLength: maxLength(10) },
      idCampoExperiencia: { required },
      idFaixaAno: { required }
    };
  }

  static validationMessages() {
    return {
      descricao: {
        required: "Descrição é obrigatória",
        maxLength: "Descrição só aceita no máximo 400 caracteres"
      },
      codigo: {
        required: "Código é obrigatório",
        maxLength: "Código só aceita no máximo 10 caracteres"
      },
      idCampoExperiencia: {
        required: "Campo Experiência é obrigatório"
      },
      idFaixaAno: {
        required: "Faixa/Ano é obrigatório"
      }
    };
  }
}
export default Objetivo;
